### PR TITLE
My code review suggestions.

### PR DESCRIPTION
**Pull request by J-Macleod**
**Description of the Project:**
The app is for keeping track of items that were given to others to borrow. There are buttons on the main screen for books, movies, and DVDs. Each gives the message that they were clicked when clicked. 

**Was the program available in UC Github on time?:**
Yes. The program was available and ready to clone when I was ready to look at and review the project.

**Is the program documented/commented well enough for you to understand?:**
The comments and documentation could have been more descriptive and specific. Also, there was some changes to comment format that would make them more in-line with coding conventions.

**Does the program compile?:**
The program did compile when I cloned it and it also compiled after I made my changes.

**Rationale behind your changes.:**
In activity_main.xml and activity_list.xml, I removed unused namespace declarations that were not necessary for the code.

In MainActivity.kt, a space before the comment's contents would be more in-line with kotlinlang.org's coding conventions.
https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace

In ExampleUnitTest.kt and ExampleInstrumentedTest.kt, I removed an empty line of multi-line comment space to better match the example on kotlinlang.org.
https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

In RetrofitClientInstance.kt, I removed code that Android Studio highlighted and says is optional and redundant, which is what kotlinlang.org recommends in this scenario.
https://kotlinlang.org/docs/coding-conventions.html#avoid-redundant-constructs

**Three specific commits that I made to my own group's GitHub repository:**
Added Menu
https://github.com/JellyV/FreeWays/commit/d72213731644070edea1935e813e487ead21adde

App Menu Buttons Added with Images
https://github.com/JellyV/FreeWays/commit/0237ef3cc76bc1633c039e52fda4df5328edeb84

Theme and Colors Applied
https://github.com/JellyV/FreeWays/commit/dae059ff7c5a9f0fb1c4349177cbe54bc6e71c9a

**Specific technical concepts I learned from the code review:**
* How to apply the lateinit keyword to a variable when there is initialization some point after construction.
* How to use the databinding library method of creating buttons for UI.
* How to move information between activities while using databinding library.

### CODE REVIEW

File: app/src/androidTest/java/com/example/mylittlelibrary/ExampleInstrumentedTest.kt [hunk 1]
```diff
@@ -10,7 +10,6 @@ import org.junit.Assert.*
 
 /**
  * Instrumented test, which will execute on an Android device.
- *
```
Comment (J-Macleod) [inline app/src/androidTest/java/com/example/mylittlelibrary/ExampleInstrumentedTest.kt:13]: Empty line removed from multi-line comment to better match the example on kotlinlang.org. https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

File: app/src/test/java/com/example/mylittlelibrary/ExampleUnitTest.kt [hunk 2]
```diff
@@ -6,7 +6,6 @@ import org.junit.Assert.*
 
 /**
  * Example local unit test, which will execute on the development machine (host).
- *
```
Comment (J-Macleod) [inline app/src/test/java/com/example/mylittlelibrary/ExampleUnitTest.kt:9]: Empty line removed from multi-line comment to better match the example on kotlinlang.org. https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
```
Comment (J-Macleod) [inline app/src/main/res/layout/activity_main.xml:4]: Unused namespace declaration removed that was not necessary for the code.

File: app/src/main/res/layout/activity_list.xml [hunk 4]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (J-Macleod) [inline app/src/main/res/layout/activity_list.xml:3]: Unused namespace declaration removed that was not necessary for the code.

File: app/src/main/java/com/example/mylittlelibrary/ui/activities/MainActivity.kt [hunk 5]
```diff
@@ -14,7 +14,7 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        //testing
+        // testing
```
Comment (J-Macleod) [inline app/src/main/java/com/example/mylittlelibrary/ui/activities/MainActivity.kt:17]: A space before the comment's contents would be more in-line with kotlinlang.org's coding conventions.
https://kotlinlang.org/docs/coding-conventions.html#horizontal-whitespace

File: app/src/main/java/com/example/mylittlelibrary/RetrofitClientInstance.kt [hunk 6]
```diff
@@ -12,7 +12,7 @@ object RetrofitClientInstance {
         get() {
             if (retrofit == null) {
                 val gson = GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss").create()
-                retrofit = retrofit2.Retrofit.Builder()
+                retrofit = Retrofit.Builder()
```
Comment (J-Macleod) [inline app/src/main/java/com/example/mylittlelibrary/RetrofitClientInstance.kt:15]: Code removed that Android Studio highlighted and says is optional and redundant, which is what kotlinlang.org says to do with optional and redundant code.
https://kotlinlang.org/docs/coding-conventions.html#avoid-redundant-constructs

File: app/src/main/java/com/example/mylittlelibrary/ui/activities/MainActivity.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/com/example/mylittlelibrary/ui/activities/MainActivity.kt:17]: Is this comment needed at all?

File: app/src/main/res/layout/activity_list.xml (hunk 4 above)
Comment (discospiff) [reply app/src/main/res/layout/activity_list.xml:3]: Are you certain this is not needed?

**Comment by discospiff** [comment]
The changes here are rather light... I'm not sure that they warrant a merge of this branch.

**Comment by andrewglanzzz** [comment]
Agree with the prof on this one, changes are rather light. I don't think this warrants a merge.

**Pull request closed without merging**